### ISSUE TITLE
Errors in French dump

### ISSUE THREAD

**Issue by yolpsoftware** [tatuylonen/wiktextract#180]
Please have a look at the inflected forms here:

https://kaikki.org/dictionary/French/meaning/f/fa/faire.html

In the rendered HTML output, everything seems to be fine. However, in the JSON, there is this:

![image](https://user-images.githubusercontent.com/1302357/201369035-85c7a1fb-e7f8-4444-9a10-2c7de958cc52.png)

Five inflected forms have an English text as their `form` property.

**Comment by kristian-clausal** [comment]
This is unfortunately because the original Wiktionary tables contain those forms in that form. The alternative is to somehow figure out what forms to ignore. There is no viable way to "translate" anything; it's Wiktionary data or nothing. On Nov 11, 2022 17:14, Yolp ***@***.***> wrote:
Please have a look at the inflected forms here:
https://kaikki.org/dictionary/French/meaning/f/fa/faire.html
In the rendered HTML output, everything seems to be fine. However, in the JSON, there is this:

Five inflected forms have an English text as their form property.

**Comment by yolpsoftware** [comment]
Ah I see, you mean this:

<img width="654" alt="image" src="https://user-images.githubusercontent.com/1302357/201395159-d815cf1f-7470-4de9-aab6-9a7acff45fa0.png">

This is a general rule in French (like in English "I have made" it is "j'ai fait", thus "present indicative of avoir + past principle").

If you aim to have a complete conjugation table in WiktExtract, you would need to look for those five specific strings and then expand them.

`present indicative of avoir + past participle`
then becomes
`j'ai fait`, `tu as fait`, `il/elle a fait`, `nous avons fait`, `vous avez fait`, `ils/elles ont fait`

This is a fairly simple rule. You just take the conjugation of `avoir` or `être` and combine them with the participle (which is in the table).

Of course this can also be done by us (and other users) downstream, but then WiktExtract would not offer complete conjugation tables.

**Comment by kristian-clausal** [comment]
We welcome any solutions to this problem.

We are a project trying to parse human-written en.wiktionary.org data; if there is something like this in the data, we're damned if we do, damned if we don't. We can't do translations because that is a genuine chicken-or-egg problem: we need to translate the tables to get the data, but to be able to translate it we need the data first! Besides which, we can't do translations in any case.

The simplest solution is to not use this data, but if we do, then people will complain that our data doesn't contain French compound tenses at all. If we strip out the "meta"-stuff in English, we're left with a bunch of {form: "avoir", "tags": "multiword-construction"...}

If someone has any really good ideas, please share it in this thread. I'll be closing it sometime in the future.

**Comment by yolpsoftware** [comment]
@kristian-clausal I understand of course that you must draw a limit somewhere, and you cannot parse / execute all kinds of "meta" stuff (commands, abbreviations and comments) that have been provided in plain English language in Wiktionary.

I just think that in this case (French compound tenses, maybe also other latin languages), the number of different cases is quite small. We have five cases for verbs that use "avoir" as their auxiliary, and five similar cases for those verbs that use "être". So it is ten special cases. I'm thinking of an algorithm of this sort:

```
if LANG == "fr":
    if "+ past participle" in item.form:
        if "avoir" in item.form:
            if "present indicative" in item.form:
                # j'ai fait, tu as fait, il/elle a fait, nous avons fait, vous avez fait, ils/elles ont fait
            else "imperfect indicative" in item.form:
                # j'avais fait, tu avais fait, il/elle avait fait, nous avions fait, vous aviez fait, ils avaient fait
            ...
        else if "être" in item.form:
            if "present indicative" in item.form:
                # je suis allé, tu es allé, ...
            ...
        else:
            # something is wrong here, maybe output a warning
```

Of course, this breaks as soon as someone changes those sources at Wiktionary, but I think that is quite unlikely. And if it breaks, we are back to where we are now (no French compounds in the output), so it definitely will not be worse than now.

> if there is something like this in the data, we're damned if we do, damned if we don't. 

I'm not sure I understand why you are saying this. What would be the problem if you parse this?

Of course, no disrespect meant. WiktExtract is a great project as it is, and those people using French compound tenses could parse those downstream just as well. If you draw a limit here and don't want to parse this, that's absolutely fine for me.

**Comment by kristian-clausal** [comment]
We "parse" wiktionary in the sense that we parse wikitext syntax and then try to guess what we need to extract from the text to put into the data. What you are suggesting is parsing *and* translating, creating *new* data that we don't have. We can't do what you suggest because it's a different kind of project; generating speculative translated text.

Your example is for "faire", but the example is found in *all* French verb conjugation tables. At best, it's going to be this one *really* elaborate subsystem just for French verb tables, and then another elaborate subsystem for Spanish ones, etc.

Not to mention this is just for en.wiktionary.com and not applicable for wiktionaries written in other languages.

I will admit we 'clean' a lot of data, like separating alternative varieties separated by /-slashes or removing unnecessary articles and qualifiers, but this is its own wild beast. It is much easier to do in post-processing language-by-language, because at that point we've gathered the 'data' needed (like what is the past participle of manger; wiktextract doesn't actually 'know' that, it just puts it in the data, and the handling of the data might be in wrong order) to do it.

Think of wiktextract as a pre-processing step. The most basic of data extractions from human-written data, just enough to get the job done so that it can be used elsewhere.

**Comment by yolpsoftware** [comment]
Ah I see, makes sense.

Do you know of any open source post-processing projects that take WiktExtract outputs and "enrich" them using some rules and heuristics?

**Comment by kristian-clausal** [comment]
Wiktextract is still pretty 'new' (in quotes) and a moving target (= annoying to use if we change anything) so I'm not sure how many projects are actually using the data in the way you mean. @tatuylonen is probably more familiar with other projects.
